src/System.Net.Http/tests/FunctionalTests/ReadOnlyMemoryContentTest.cs

```diff
@@ -0,0 +1,329 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Buffers;
+using System.IO;
+using System.IO.Tests;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace System.Net.Http.Functional.Tests
+{
+    public class ReadOnlyMemoryContentTest
+    {
+        [Theory]
+        [InlineData(0, false)]
+        [InlineData(1, false)]
+        [InlineData(1024, false)]
+        [InlineData(0, true)]
+        [InlineData(1, true)]
+        [InlineData(1024, true)]
+        public void ContentLength_LengthMatchesArrayLength(int contentLength, bool useArray)
+        {
+            Memory<byte> memory;
+            OwnedMemory<byte> ownedMemory;
+            ReadOnlyMemoryContent content = CreateContent(contentLength, useArray, out memory, out ownedMemory);
+
+            Assert.Equal(contentLength, content.Headers.ContentLength);
+
+            ownedMemory?.Dispose();
+        }
+
+        [Theory]
+        [InlineData(false)]
+        [InlineData(true)]
+        public async Task ReadAsStreamAsync_TrivialMembersHaveExpectedValuesAndBehavior(bool useArray)
+        {
+            const int ContentLength = 42;
+            Memory<byte> memory;
+            OwnedMemory<byte> ownedMemory;
+            ReadOnlyMemoryContent content = CreateContent(ContentLength, useArray, out memory, out ownedMemory);
+
+            using (Stream stream = await content.ReadAsStreamAsync())
+            {
+                // property values
+                Assert.Equal(ContentLength, stream.Length);
+                Assert.Equal(0, stream.Position);
+                Assert.True(stream.CanRead);
+                Assert.True(stream.CanSeek);
+                Assert.False(stream.CanWrite);
+
+                // not supported
+                Assert.Throws<NotSupportedException>(() => stream.SetLength(42));
```

Comment:
Nit: use the constant

Reply:
Ah, nevermind, I see the value here doesn't matter.  Thought you were setting position.